src/anchor-navigation/internal.tsx

```diff
@@ -0,0 +1,112 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useCallback, useEffect, useMemo } from 'react';
+import clsx from 'clsx';
+import styles from './styles.css.js';
+import { AnchorNavigationProps } from './interfaces';
+import { checkSafeUrl } from '../internal/utils/check-safe-url';
+import useScrollSpy from './scroll-spy.js';
+import { fireCancelableEvent, fireNonCancelableEvent, isPlainLeftClick } from '../internal/events/index';
+import { useStableCallback } from '@cloudscape-design/component-toolkit/internal';
+import { InternalBaseComponentProps } from '../internal/hooks/use-base-component/index.js';
+import { getBaseProps } from '../internal/base-component/index.js';
+
+export default function InternalAnchorNavigation({
+  anchors,
+  ariaLabelledby,
+  onFollow,
+  onActiveHrefChange,
+  activeHref = '',
+  __internalRootRef = null,
+  scrollSpyOffset = 0,
+  ...props
+}: AnchorNavigationProps & InternalBaseComponentProps) {
+  const baseProps = getBaseProps(props);
+
+  const hrefs = useMemo(() => anchors.map(anchor => anchor.href), [anchors]);
+
+  const onFollowHandler = useCallback(
+    (anchor: AnchorNavigationProps.Anchor, sourceEvent: React.SyntheticEvent | Event) => {
+      fireCancelableEvent(onFollow, anchor, sourceEvent);
+    },
+    [onFollow]
+  );
+
+  const onActiveHrefChangeHandler = useStableCallback((newActiveAnchor: AnchorNavigationProps.Anchor | undefined) => {
```

Comment:
Is this function necessary? Could we just call `fireNonCancellableEvent` directly on line 47 below instead?